### DISCUSSION TITLE
How to manually lower the commit_fee for channels?

### DISCUSSION THREAD

**Discussion by ShookLyngs** [lightningnetwork/lnd#9203] [Q&A]
Say A opens a channel of 20,000 sats to B and pushes 1,000 sats to B. Since the current fee rate on testnet is high, A now has 785 sats in balance, while B has 1,000. The latest `commit_fee` is 17,555 sats.

The problem is that even though A seems to have sufficient balance for a 1 sat transfer to B (or conversely, B should be able to transfer 1 sat to A), the transfer fails with the following error:

```
FAILED, reason: FAILURE_REASON_INSUFFICIENT_BALANCE
```

A then tries to update the `lnd.conf` to the following, even providing a static JSON via the `fee.url` parameter:

```
max-channel-fee-allocation=0.1
max-commit-fee-rate-anchors=1
fee.url=https://raw.githubusercontent.com/ShookLyngs/test-lnd-fee-url/refs/heads/main/static-fees.json
fee.min-update-timeout=1m
fee.max-update-timeout=1m1s
```

However, the `commit_fee` remains high at the original value of 17,555. How can A force an update to the `commit_fee` except waiting for the network fees to drop, for example, setting it to a very low value so that it doesn't block the entire transaction process? By the way, A is using `bitcoind` as the backend node.

**Comment by guggero** [comment]
What channel type is the channel in question?

**Reply by ShookLyngs** [reply]
> What channel type is the channel in question?

It's an `ANCHORS` type channel. In theory with the static `fee.url` set, providing `fee_by_block_target` and `min_relay_feerate` as low as `1012`, the fee rate should have been lowered by the random timeout `update_fee` event. However, in reality, the channel was not updated for a long time (about a day or so). Here's an error related to the channel; I suppose it's about updating the fee:

```
[WRN] LNWL: ChannelPoint(x:0): Set available amount to 0 because we could not pay for the CommitmentFee of the new ChannelState: ourBalance is negative after applying the fee: ourBalance=17986000, current commitFee(w/o additional htlc)=17555000 mSAT, feeBuffer=20241000 mSAT (type=additionalhtlc) local_chan_initiator
```

I then deposited more coins into A's L1 address and opened another channel with the `lnd.conf` mentioned above (static `fee.url` and more, just to limit the `commit_fee`). This time, when the new channel was opened, both channels' fee rate was updated, the new channel's fee rate was `250`, the old one was `253`. 

While the old channel was able to facilitate transfers between A and B, the new channel, on the other hand, was unable to make transfers because its fee rate was `250`, which was too low to be accepted. This warning appears in the logs:

```
[WRN] CRTR: ShortChannelID=x: cannot add outgoing htlc: commitment fee per kw 250 sat/kw below fee floor 253 sat/kw
```
